### ISSUE TITLE
Support more disposable patterns

### ISSUE THREAD

**Issue by vpenades** [dotnet/roslyn-analyzers#6019]
### Analyzer

**Diagnostic ID**: [CA2213](https://docs.microsoft.com/es-es/dotnet/fundamentals/code-analysis/quality-rules/ca2213)

### Describe the improvement

Right now, the analyzer reports this as a false positive:

```c#
Interlocked.Exchange(ref disposable, null)?.Dispose();
```

### Describe suggestions on how to achieve the rule

This disposing pattern should not trigger a false positive.

### Additional context

This way of disposing an object is the safest one, because it prevents the object's Dispose to be called two times in case it is mistakenly called from two threads at the same time... chances of collisions might be small but not zero, which would make an application crashing from time to time, in a bug that is nearly impossible to detect.

So this pattern not only needs to stop giving false positives, but also to be encouraged by the analyzer, by giving the suggestion to switch to it in case it finds the classic `if (disposable != null) disposable.Dispose();`
